lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -209,7 +209,9 @@ def _execute_win(get_file_cmd)
   end
 
   def _execute_nix(get_file_cmd)
-    return _generate_fileless(get_file_cmd) if datastore['FETCH_FILELESS']
+    return _generate_fileless(get_file_cmd) if datastore['FETCH_FILELESS'] == 'bash'
+    return _generate_fileless_python(get_file_cmd) if datastore['FETCH_FILELESS'] == 'python3.8'
```

Comment:
```suggestion
    return _generate_fileless_python(get_file_cmd) if datastore['FETCH_FILELESS'] == 'python3.8+'
```
Probably worthwhile to specify this should work with future python, too.